### PR TITLE
ramips: cleanup ralink SPI driver

### PR DESCRIPTION
**Pull request by neheb**
Increase usage of devm to get rid of goto and _remove.

Get rid of hw_reset_count. It's not really used for anything.

Use dev_err_probe to handle potential EPROBE_DEFER.

### CODE REVIEW

**Review by DragonBluep — APPROVED** [review]
`Tested-by: Shiji Yang <yangshiji66@qq.com>` on MT7620.

**Comment by neheb** [comment]
ping @robimarko @hauke

File: target/linux/ramips/patches-6.6/821-SPI-ralink-add-Ralink-SoC-spi-driver.patch [hunk 1]
```diff
@@ -507,50 +496,12 @@ Acked-by: John Crispin <blogic@openwrt.org>
...
 +
-+	dev_set_drvdata(&pdev->dev, master);
-+
 +	rs = spi_master_get_devdata(master);
 +	rs->master = master;
 +	rs->base = base;
 +	rs->clk = clk;
 +
-+	if (atomic_inc_return(&hw_reset_count) == 1) {
-+		ret = device_reset(&pdev->dev);
```
Comment (hauke) [inline target/linux/ramips/patches-6.6/821-SPI-ralink-add-Ralink-SoC-spi-driver.patch:518]: I think this code is used to call the `device_reset` only for the first instance of this device. Probably multiple SPI cores  share the reset line and the driver wants to do a HW reset before the first instance is used. 

I think you can call it always, if the reset is shared it is only executed for the first device, see:
https://elixir.bootlin.com/linux/v6.6.53/source/drivers/reset/core.c#L327

Comment (neheb) [reply target/linux/ramips/patches-6.6/821-SPI-ralink-add-Ralink-SoC-spi-driver.patch:518]: That's a different function, no?

Comment (hauke) [reply target/linux/ramips/patches-6.6/821-SPI-ralink-add-Ralink-SoC-spi-driver.patch:518]: In the end `device_reset` calls `reset_control_reset`. I think you can remove the `hw_reset_count` handling and just always call `device_reset(&pdev->dev);`.

**Comment by neheb** [comment]
@hauke added back device_reset

**Pull request merged**